src/kakarot/instructions/system_operations.cairo

```diff
@@ -131,7 +132,54 @@ namespace SystemOperations {
         with_attr error_message("Kakarot: Reverted with reason: {revert_reason}") {
             assert TRUE = FALSE;
         }
+        // TODO: this is never reached, raising with cairo prevent from implementing a true REVERT
+        // TODO: that still returns some data. This is especially problematic for sub contexts.
         let ctx = ExecutionContext.update_memory(self=ctx, new_memory=memory);
         return ctx;
     }
+
+    // @notice CALL operation.
+    // @dev
+    // @custom:since Frontier
+    // @custom:group System Operations
+    // @custom:gas 0 + dynamic gas
+    // @custom:stack_consumed_elements 7
+    // @custom:stack_produced_elements 1
+    // @return The pointer to the sub context.
+    func exec_call{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        alloc_locals;
+        let (stack, popped) = Stack.pop_n(self=ctx.stack, n=7);
+        let ctx = ExecutionContext.update_stack(ctx, stack);
+
+        let gas = 2 ** 128 * popped[6].high + popped[6].low;
```

Comment:
is this the safest way we can convert a Uint256 to a felt?

Reply:
If that the intention can we wrap it a utility function or add a comment to ease maintenability?

Reply:
> is this the safest way we can convert a Uint256 to a felt?

I don't know but I've not found a Uint256 to felt util. Are you scared about a potential overflow?